From 9dd43182ecb9f8406a1aecd0719f2c5225d3101e Mon Sep 17 00:00:00 2001
From: Carlos Garnacho <carlosg@gnome.org>
Date: Thu, 28 Nov 2019 15:47:02 +0100
Subject: [PATCH 1/3] sound: Ensure to preserve sound theme when changing from
 default

The sound theme change itself triggers signals that are handled here
in the GSettings handlers, possibly resulting in it changing the
theme back to default. Ensure we are handling a real change here,
the sound theme will be updated through other means (eg. the row
changed handler).
---
 panels/sound/gvc-sound-theme-chooser.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/panels/sound/gvc-sound-theme-chooser.c b/panels/sound/gvc-sound-theme-chooser.c
index 93eeb155c..9d1051f04 100644
--- a/panels/sound/gvc-sound-theme-chooser.c
+++ b/panels/sound/gvc-sound-theme-chooser.c
@@ -632,11 +632,11 @@ update_theme (GvcSoundThemeChooser *chooser)
                         load_theme_name (DEFAULT_THEME,
                                          &chooser->current_parent);
                 }
+
+                update_alerts_from_theme_name (chooser, chooser->current_theme);
         }
 
         gtk_widget_set_sensitive (chooser->selection_box, events_enabled);
-
-        update_alerts_from_theme_name (chooser, chooser->current_theme);
 }
 
 static GObject *
-- 
2.24.0

